bin/agent-data-plane/src/main.rs

```diff
@@ -176,6 +177,7 @@ fn create_topology(
     let origin_enrichment_config = OriginEnrichmentConfiguration::from_configuration(configuration)
         .error_context("Failed to configure origin enrichment transform.")?
         .with_environment_provider(env_provider);
+    let name_filter_configuration = DogstatsDPrefixFilterConfiguration::from_configuration(configuration)?;
```

Comment:
Probably change all the instance of `name_filter` (both this variable name and the component ID below) to something like `dsd_prefix_filter`?